topo_processor/stac/collection.py

```diff
@@ -15,12 +16,14 @@ class Collection:
     description: str
     license: str
     data_type: DataType
+    temp_dir: str
     items: List["Item"]
     providers: List[stac.Provider]
     collection_output_path: str
 
     def __init__(self, data_type: DataType):
         self.data_type = data_type
+        self.temp_dir = mkdtemp()
```

Comment:
I don't think this logic should live here, this logic is around the collection not around temporary processing of it.